lib/semian/mysql2.rb

```diff
@@ -83,6 +83,23 @@ def query(*args)
       end
     end
 
+    def with_resource_timeout(temp_timeout)
```

Comment:
Should we support different timeouts for read / write, or is it assumed this timeout is simply for the recovery phase so there is no need for it to be fine grained?